cloud/pkg/edgecontroller/manager/pod.go

```diff
@@ -2,36 +2,25 @@ package manager
 
 import (
 	"reflect"
-	"sync"
 
 	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
 	"github.com/kubeedge/kubeedge/pkg/apis/componentconfig/cloudcore/v1alpha1"
 )
 
-// CachePod is the struct save pod data for check pod is really changed
-type CachePod struct {
-	metav1.ObjectMeta
-	Spec v1.PodSpec
-}
-
 // PodManager is a manager watch pod change event
 type PodManager struct {
 	// events from watch kubernetes api server
 	realEvents chan watch.Event
 
 	// events merged
 	mergedEvents chan watch.Event
-
-	// pods, key is UID, value is *v1.Pod
-	pods sync.Map
 }
 
-func (pm *PodManager) isPodUpdated(old *CachePod, new *v1.Pod) bool {
+func isPodUpdated(old *v1.Pod, new *v1.Pod) bool {
 	// does not care fields
 	old.ObjectMeta.ResourceVersion = new.ObjectMeta.ResourceVersion
 	old.ObjectMeta.Generation = new.ObjectMeta.Generation
```

Comment:
from k8s description, we cannot modify the resource from event, seems modifying the old value is not right

Reply:
update the parameter to struct type and we will not update the filed in the cache